community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/IdGeneratorImpl.java

```diff
@@ -155,20 +156,11 @@ public synchronized long nextId()
             // special values, f.ex. the end of a relationships/property chain.
             id = highId.incrementAndGet();
         }
-        assertIdWithinCapacity( id );
+        IdValidator.assertValidId( id, max );
```

Comment:
@tinwelint fixed now. also made sure that other code uses `IdValidator` instead of directly checking `INTEGER_MINUS_ONE`.